src/v/cloud_storage_clients/abs_client.cc

```diff
@@ -547,6 +547,45 @@ ss::future<> abs_client::do_delete_object(
     }
 }
 
+ss::future<result<abs_client::delete_objects_result, error_outcome>>
+abs_client::delete_objects(
+  const bucket_name& bucket,
+  std::vector<object_key> keys,
+  ss::lowres_clock::duration timeout) {
+    std::vector<ss::future<result<abs_client::no_response, error_outcome>>>
+      delete_futs;
+    delete_futs.reserve(keys.size());
+
+    for (const auto& key : keys) {
+        delete_futs.push_back(delete_object(bucket, key, timeout));
+    }
```

Comment:
Should we limit parallelism here somehow? For instance, limit the number of futures of deletion futures to 100 at any given time.

Reply:
Yes, I will add `max_concurrent_for_each` here

Reply:
Changed to do 32 operations at a time. I found it tricky to come up with a good limit here for the network calls. 

Used this number because I see it in use as a limit in many other max concurrent calls across the tree, but open to other suggestions.

Reply:
I would have gone for a slightly higher number, but I don't think it matters too much. We can always adjust if need be.